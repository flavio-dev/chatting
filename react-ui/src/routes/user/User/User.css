.UserContent {
  margin-left: auto;
  margin-right: auto;
  width: 95%;
  padding: 40px 0; }
  @media only screen and (min-width: 601px) {
    .UserContent {
      width: 90%; } }
  @media only screen and (min-width: 601px) {
    .UserContent {
      padding: 60px 0; } }

.UserTitle {
  color: #3A2424;
  font-family: "Lobster", Helvetica, sans-serif;
  font-size: 40px;
  margin-bottom: 60px;
  text-align: center; }
  @media only screen and (min-width: 601px) {
    .UserTitle {
      font-size: 60px;
      margin-bottom: 40px; } }

.UserBoard {
  display: flex;
  flex-wrap: wrap; }
