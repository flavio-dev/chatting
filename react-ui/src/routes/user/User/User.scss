@import "globals.scss";

.UserLayout {
  // this should be part of a different Component called BoardLayout which sole purpuse would
  // have been to deal with the rendering of the SidebarContainer and UserContainer
  display: flex;
}

.UserCol1 {
  background-color: $color-milk-chocolate;
  width: 30%;
}

.UserContent {
  @include default-wrapper-width();
  padding: 40px 0;
  width: 70%;

  @media #{$medium-and-up} {
    padding: 60px 0;
  }
}

.UserTitle {
  color: $color-chocolate;
  font-family: $title-font-family;
  font-size: 40px;
  margin-bottom: 60px;
  text-align: center;
  text-transform: capitalize;

  @media #{$medium-and-up} {
    font-size: 60px;
    margin-bottom: 40px;
  }
}

.UserBoard {
  display: flex;
  flex-wrap: wrap;
}
