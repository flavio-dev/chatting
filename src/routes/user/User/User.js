import React, { Component } from 'react'
import PropTypes from 'prop-types'

import UserWindow from 'components/UserWindow'

import styles from './User.css'

class User extends Component {
  constructor(props) {
    super(props)
    this.userId = ''
    this.state = {
      messages: this.props.messages,
      users: this.props.users
    }

    this.filterMessage = this.filterMessage.bind(this)
  }

  componentWillMount() {
    this.userId = this.props.match.params.userId
    this.props.setUserId(this.userId)
  }

  componentDidMount() {
    this.props.setConnection(this.userId)
    this.props.getInitialListUsers()
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.messages) {
      this.setState({
        messages: nextProps.messages,
        users: nextProps.users
      })
    }
  }

  filterMessage(user) {
    return this.state.messages.filter(message => {
      return (message.from === user && message.to === this.userId) ||
        (message.from === this.userId && message.to === user)
    })
  }

  filterMessageAll() {
    console.log('filterMessageAll = ', this.state.messages)
    return this.state.messages.filter(message => {
      return message.to === 'ALL'
    })
  }

  render() {
    return (
      <div className={styles.UserContent}>
        <h2 className={styles.UserTitle}>Hello {this.userId}</h2>
        <div className={styles.UserBoard}>
          <UserWindow
            title='forum'
            messages={this.filterMessageAll()}
            to='ALL'
            you={this.userId}
            sendMessage={this.props.sendMessage}
           />
          {this.state.users.map((user, index) => {
            return <UserWindow
              key={index}
              title={user}
              messages={this.filterMessage(user)}
              to={user}
              you={this.userId}
              sendMessage={this.props.sendMessage}
              online
            />
          })}
        </div>
      </div>
    )
  }
}

User.propTypes = {
  match: PropTypes.object,
  setUserId: PropTypes.func,
  setConnection: PropTypes.func,
  sendMessage: PropTypes.func,
  getInitialListUsers: PropTypes.func,
  messages: PropTypes.arrayOf(PropTypes.shape({
    message: PropTypes.string,
    from: PropTypes.string,
    to: PropTypes.string
  })),
  users: PropTypes.array
}

export default User
