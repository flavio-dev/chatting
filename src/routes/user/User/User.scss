@import "globals.scss";

.UserContent {
  @include default-wrapper-width();
}

.UserTitle {
  color: $color-chocolate;
  font-family: $title-font-family;
  font-size: 40px;
  margin-bottom: 60px;
  text-align: center;

  @media #{$medium-and-up} {
    font-size: 60px;
    margin-bottom: 40px;
  }
}

.UserBoard {
  display: flex;
}
