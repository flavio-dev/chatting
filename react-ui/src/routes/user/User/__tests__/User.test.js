import React from 'react'
import User from '../User'
import { shallow, mount, setState, render } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Testing User.js', () => {
  it('renders correctly with empty list of messages/users', () => {
    const setUserId = jest.fn()
    const setConnection = jest.fn()
    const sendMessage = jest.fn()
    const getInitialListUsers = jest.fn(() => [])
    const match = {
      params: {
        userId: 'userId'
      }
    }
    const tree = renderer.create(
      <User
        users={[]}
        messages={[]}
        match={match}
        setUserId={setUserId}
        setConnection={setConnection}
        sendMessage={sendMessage}
        getInitialListUsers={getInitialListUsers}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with empty list of messages but with users', () => {
    const setUserId = jest.fn()
    const setConnection = jest.fn()
    const sendMessage = jest.fn()
    const getInitialListUsers = jest.fn(() => [])
    const match = {
      params: {
        userId: 'userId'
      }
    }
    const tree = renderer.create(
      <User
        users={['secondUser', 'thirdUser']}
        messages={[]}
        match={match}
        setUserId={setUserId}
        setConnection={setConnection}
        sendMessage={sendMessage}
        getInitialListUsers={getInitialListUsers}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with list of users and list of messages', () => {
    const setUserId = jest.fn()
    const setConnection = jest.fn()
    const sendMessage = jest.fn()
    const getInitialListUsers = jest.fn(() => [])
    const match = {
      params: {
        userId: 'userId'
      }
    }
    const tree = renderer.create(
      <User
        users={['secondUser', 'thirdUser']}
        messages={[{
          from: 'userId',
          to: 'secondUser',
          message: 'the message'
        }]}
        match={match}
        setUserId={setUserId}
        setConnection={setConnection}
        sendMessage={sendMessage}
        getInitialListUsers={getInitialListUsers}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with list of users and list of messages to ALL', () => {
    const setUserId = jest.fn()
    const setConnection = jest.fn()
    const sendMessage = jest.fn()
    const getInitialListUsers = jest.fn(() => [])
    const match = {
      params: {
        userId: 'userId'
      }
    }
    const tree = renderer.create(
      <User
        users={['secondUser', 'thirdUser']}
        messages={[{
          from: 'userId',
          to: 'ALL',
          message: 'the message'
        }]}
        match={match}
        setUserId={setUserId}
        setConnection={setConnection}
        sendMessage={sendMessage}
        getInitialListUsers={getInitialListUsers}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
