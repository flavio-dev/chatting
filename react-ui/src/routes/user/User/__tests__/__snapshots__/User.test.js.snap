// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing User.js renders correctly with empty list of messages but with users 1`] = `
<div
  className="UserContent"
>
  <h2
    className="UserTitle"
  >
    Hello 
    userId
  </h2>
  <div
    className="UserBoard"
  >
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
         
        forum
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          />
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
        <div
          className="WindowOnline"
        />
         
        secondUser
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          />
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
        <div
          className="WindowOnline"
        />
         
        thirdUser
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          />
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing User.js renders correctly with empty list of messages/users 1`] = `
<div
  className="UserContent"
>
  <h2
    className="UserTitle"
  >
    Hello 
    userId
  </h2>
  <div
    className="UserBoard"
  >
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
         
        forum
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          />
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing User.js renders correctly with list of users and list of messages 1`] = `
<div
  className="UserContent"
>
  <h2
    className="UserTitle"
  >
    Hello 
    userId
  </h2>
  <div
    className="UserBoard"
  >
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
         
        forum
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          />
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
        <div
          className="WindowOnline"
        />
         
        secondUser
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          >
            <div
              className="WindowYou"
            >
              <span>
                you: 
              </span>
              the message
            </div>
          </div>
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
        <div
          className="WindowOnline"
        />
         
        thirdUser
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          />
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing User.js renders correctly with list of users and list of messages to ALL 1`] = `
<div
  className="UserContent"
>
  <h2
    className="UserTitle"
  >
    Hello 
    userId
  </h2>
  <div
    className="UserBoard"
  >
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
         
        forum
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          >
            <div
              className="WindowYou"
            >
              <span>
                you: 
              </span>
              the message
            </div>
          </div>
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
        <div
          className="WindowOnline"
        />
         
        secondUser
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          />
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
    <div
      className="WindowWrapper"
    >
      <h2
        className="WindowTitle"
      >
        <div
          className="WindowOnline"
        />
         
        thirdUser
      </h2>
      <div
        className="Window"
      >
        <div
          className="WindowContentWrapper"
        >
          <div
            className="WindowContent"
          />
        </div>
        <div
          className="WindowCTA"
        >
          <textarea
            onChange={[Function]}
            placeholder="Type a message..."
            value=""
          />
          <button
            onClick={[Function]}
          >
            Send
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
